netket/utils/history.py

```diff
@@ -14,6 +14,29 @@
 
 import numpy as np
 from numbers import Number
+import jax.numpy as jnp
+
+
+def _is_scalar(val):
+    """
+    Returns true if the input is a scalar-like object.
+
+    Checks whever it is a Python's number, or any numpy-API
+    scalar.
+    """
+    if isinstance(val, Number):
+        return True
+    elif np.isscalar(val):
+        return True
+    elif jnp.isscalar(val):
+        return True
```

Comment:
Hm, I was thinking that this could be simplified... but numpy / jax behavior is indeed different:
```python
In [11]: a
Out[11]: DeviceArray(1., dtype=float32)

In [12]: np.isscalar(a)
Out[12]: False

In [13]: jnp.isscalar(a)
Out[13]: True

In [14]: a = np.array(1.)

In [15]: np.isscalar(a)
Out[15]: False

In [16]: jnp.isscalar(a)
Out[16]: False
```

Reply:
I think it's time to write a function using @dispatch to do this